### PR TITLE
Ezhil - language support in ACE editor

### PR DESCRIPTION
**Pull request by arcturusannamalai**
Ezhil language is a Tamil script based programming language. Ezhil has notably been accepted for syntax highlighting with Python Pygments project, among other syntax highlighting support libraries.
See also: www.ezhillang.org

This PR adds support for Ezhil syntax highlighting in the ACE editor.

### CODE REVIEW

File: lib/ace/mode/ezhil.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+    this.autoOutdent = function(state, doc, row) {
+        // outdenting in python is slightly different because it always applies
+        // to the next line and only of a new line is inserted
+        
+        row += 1;
+        var indent = this.$getIndent(doc.getLine(row));
+        var tab = doc.getTabString();
+        if (indent.slice(-tab.length) == tab)
+            doc.remove(new Range(row, indent.length-tab.length, row, indent.length));
```
Comment (nightwing) [inline lib/ace/mode/ezhil.js]: Range is not defined see https://travis-ci.org/ajaxorg/ace/builds/97144623

File: lib/ace/mode/ezhil.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
+/*
+(C) 2013 Ezhil Language Project
+(C) 2013 Sridhar Ganesan
+*/
```
Comment (nightwing) [inline lib/ace/mode/ezhil.js]: To merge this we need a signed [CLA](https://github.com/ajaxorg/ace/blob/master/CONTRIBUTING.md), and BSD licence similar to [other modes](https://github.com/ajaxorg/ace/blob/master/lib/ace/mode/javascript.js#L1-L29).

File: lib/ace/ext/modelist.js [hunk 3]
```diff
@@ -74,6 +74,7 @@ var supportedModes = {
     Elixir:      ["ex|exs"],
     Elm:         ["elm"],
     Erlang:      ["erl|hrl"],
+    Ezhil:       ["ezhil"],
```
Comment (nightwing) [inline lib/ace/ext/modelist.js]: is the default file extension `n` or `ezhil`?

File: demo/kitchen-sink/doclist.js [hunk 4]
```diff
@@ -66,8 +66,32 @@ function makeHuge(txt) {
...
+    "docs/coffeescript.coffee": "CoffeeScript",
+    "docs/coldfusion.cfm": "ColdFusion",
+    "docs/cpp.cpp": "C/C++",
+    "docs/csharp.cs": "C#",
+    "docs/css.css": "CSS",
+    "docs/curly.curly": "Curly",
+    "docs/dart.dart": "Dart",
+    "docs/diff.diff": "Diff",
+    "docs/dot.dot": "Dot",
+    "docs/ezhil.n": "Ezhil",
```
Comment (nightwing) [inline demo/kitchen-sink/doclist.js]: Docs are added automatically based on modelist, so this file doesn't need to be changed

**Comment by arcturusannamalai** [comment]
@nightwing - thanks for your comments; I will need a few days to a week to update the code and follow through on the updated changeset.

**Comment by arcturusannamalai** [comment]
I have executed the CLA. Will work on the rest of the updates soon.

**Comment by arcturusannamalai** [comment]
Give up on PR. 🤣

**Pull request closed without merging**